analyzers/src/SonarAnalyzer.CFG/ControlFlowGraph/CSharpControlFlowGraphBuilder.cs

```diff
@@ -500,7 +500,11 @@ private Block BuildExpression(ExpressionSyntax expression, Block currentBlock)
         private static bool IsTooComplex(SyntaxNode node)
         {
             var count = 0;  // Limit descending for performance reasons
-            return node.DescendantNodes(x => ++count < SupportedExpressionNodeCountLimit).Count() >= SupportedExpressionNodeCountLimit;
+            // The evaluation doesn't descend into the content of lambda expression. We need to tolerate these for Razor Layout pages.
```

Comment:
>The evaluation doesn't descend into the content of lambda expression

Where this logic happens? Is it in this builder as well, or a different class? I'm just thinking ahead if this may change in the future, the evaluation may descend into lambas, and not to forget about this IsTooComplex.

I suggest to add there a comment that `IsTooComplex` should be revisited if we decide to descent into lambda expressions in the future.

Reply:
Or is the comment just an explanation for the logic below?

Reply:
It's related to the fact, that we don't support lambdas and we don't evaluate their inner behavior. 
This `IsTooComplex` problem should go away by (re)design to Roslyn CFG so we don't need to put too much attention to it.